File: GUI/grid_gui/SmartSensor.cpp
#include "SmartSensor.h"

//constructors , destructors
SmartSensor::SmartSensor()
{

}
SmartSensor::SmartSensor(uint64_t id, const std::pair<double, double> &positionGPS,
	double range, const std::vector<std::pair<std::string, double> > &measurments) :
	m_id(id),
	m_positionGPS(positionGPS),
	m_measurments(measurments),
	m_range(range)
{

}

SmartSensor::~SmartSensor()
{

}

//setters
void SmartSensor::setSensorPositionXY(double pos1, double pos2)
{
	m_positionXY = std::make_pair(pos1, pos2);
}
void SmartSensor::setSensorPositionGPS(const std::pair<double, double> &pos)
{
	m_positionGPS = pos;
}

void SmartSensor::setSensorId(uint64_t id)
{
	m_id = id;
}

void SmartSensor::setSensorRadius(double radius)
{
	m_range = radius;
}

std::pair<double, double> SmartSensor::getPositionGPS()const
{
	return m_positionGPS;
}

std::pair<double, double> SmartSensor::getPositionXY()const
{
	return m_positionXY;
}

double SmartSensor::getPositionLong()const
{
	return m_positionGPS.second;
}

double SmartSensor::getPositionLat()const
{
	return m_positionGPS.first;
}

uint64_t SmartSensor::getId()const
{
	return m_id;
}

double SmartSensor::getRange()const
{
	return m_range;
}

File: Java_Gui/src/SmartSensor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package digimesh.xbee.gui;

import java.util.ArrayList;
import java.util.List;


public class SmartSensor {
        public ArrayList<Measurment>  m_measurments;
	private String m_id;
	private Position m_positionGPS; // <latitude, longitude>
	private Position m_positionXY; // position on map image
	private double m_range; // in meters - sensor range 
        public boolean sleepEnable;
        private double sleepTime;
        public boolean setPositionGPS;
        public boolean setPositionXY;
        private int measurmentToDraw;
        public boolean hasMeasurements;
        
	public SmartSensor()
        {
            
        }
        
        public SmartSensor(String id)
        {
            m_id = id;
            m_positionGPS = new Position();
            m_range = 0;
            m_measurments = new ArrayList<Measurment>();
            sleepEnable = false;
            sleepTime = 0;
            setPositionGPS = false;
            setPositionXY = false;
            measurmentToDraw = 0;
            hasMeasurements = false;
        }

        public SmartSensor(String id, Position positionGPS, double range)
        {
            m_id = id;
            m_positionGPS = positionGPS;
            m_range = range;
            sleepEnable = false;
            sleepTime = 0;
            setPositionGPS = true;
            setPositionXY = false;
            measurmentToDraw = 0;
            hasMeasurements = false;
        }
        
	public SmartSensor(String id, Position positionGPS, double range, ArrayList<Measurment> measurments)
        {
            m_id = id;
            m_positionGPS = positionGPS;
            m_range = range;
            m_measurments = measurments;
            sleepEnable = false;
            sleepTime = 0;
            setPositionGPS = true;
            setPositionXY = false;
            measurmentToDraw = 0;
            hasMeasurements = true;
        }

	//setters
	public void setSensorPositionXY(double pos1, double pos2)
        {
            m_positionXY = new Position(pos1, pos2);
            setPositionXY = true;
            
        }
        
	public void setSensorPositionGPS(Position pos)
        {
            m_positionGPS = pos;
            setPositionGPS = true;
        }
        
	public void setSensorId(String id)
        {
            m_id = id;
        }
        
	public void setSensorRadius(double radius)
        {
            m_range = radius;
        }
        
        public void setSensorSleepTime(double sleepTime)
        {
            if(sleepTime > 0)
            {
                sleepEnable = true;
            }
            else
            {
                sleepEnable = false;
            }
            this.sleepTime = sleepTime;          
        }
        
        public void setMeasurmentToDraw(int nr)
        {
            measurmentToDraw = nr;
        }
        
	//getters
	public Position getPositionGPS()
        {
            return m_positionGPS;
        }
        
	public Position getPositionXY()
        {
           return m_positionXY; 
        }
        
	public double getPositionLong()
        {
            return m_positionGPS.positionY;
        }
        
	public double getPositionLat()
        {
            return m_positionGPS.positionX;
        }
	String getId()
        {
            return m_id;
        }
	
        double getRange()
        {
            return m_range;
        }
        
        double getSleepTime()
        {
            return sleepTime;
        }
        
        int getMeasurmentToDraw()
        {
            return measurmentToDraw;
        }
}
